ath79: add support for MikroTik RouterBOARD 921GS-5HPacD-15s (mANTBox 15s)

SvenRoederer opened the pull request:
This ports the board from ar71xx.

I was only able to boot to initrd ~~, but writing to flash seems to fail. As I don't have a serial console attached, I was not able to look at this~~ and write to flash. But I already like to share for review, comments and test. @rogerpueyo

As the embedded RB921-pcb is a stripped down version of the RB922 this patch adds a common dtsi for this series and includes this to the final dts-files.

See https://mikrotik.com/product/RB921GS-5HPacD-15S for more info.

Specifications:
 - SoC: Qualcomm Atheros QCA9558 (720 MHz)
 - RAM: 128 MB
 - Storage: 128 MB NAND
 - Wireless: external QCA9892 802.11a/ac 2x2:2
 - Ethernet: 1x 1000/100/10 Mbps, integrated, via AR8031 PHY, passive PoE in
 - SFP: 1x host

Working:
 - NAND storage detection
 - Ethernet
 - Wireless
 - Sysupgrade
 - 1x user LED (blinks during boot, sysupgrade)
 - Reset button

Untested:
 - SFP cage (probably not working)

SvenRoederer committed:
ath79: add support for MikroTik RouterBOARD 921GS-5HPacD-15s

The MikroTik RouterBOARD 921GS-5HPacD-15s (mANTBox 15s) is an outdoor
antenna for 5 GHz with an built-in router. This ports the board from
ar71xx.

See https://mikrotik.com/product/RB921GS-5HPacD-15S for more info.

Specifications:
 - SoC: Qualcomm Atheros QCA9558 (720 MHz)
 - RAM: 128 MB
 - Storage: 128 MB NAND
 - Wireless: external QCA9892 802.11a/ac 2x2:2
 - Ethernet: 1x 1000/100/10 Mbps, integrated, via AR8031 PHY, passive PoE in
 - SFP: 1x host

Working:
 - NAND storage detection
 - Ethernet
 - Wireless
 - 1x user LED (blinks during boot, sysupgrade)
 - Reset button
 - Sysupgrade

Untested:
 - SFP cage (probably not working)

Installation (untested):
 - Boot initramfs image via TFTP and then flash sysupgrade image

As the embedded RB921-pcb is a stripped down version of the RB922 this patch
adds a common dtsi for this series and includes this to the final dts-files.

Signed-off-by: Sven Roederer <devel-sven@geroedel.de>

adrianschmutzler commented on target/linux/ath79/dts/qca9558_mikrotik_routerboard-92x.dtsi:
```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
```
No dts-v1 in DTSI, only in DTS, please.

SvenRoederer replied:
Thanks, missed that during copying ...

adrianschmutzler commented on target/linux/ath79/dts/qca9558_mikrotik_routerboard-92x.dtsi:
```diff
@@ -0,0 +1,156 @@
+
+/ {
+	compatible = "mikrotik,routerboard-922uags-5hpacd", "qca,qca9558";
+	model = "MikroTik RouterBOARD 922UAGS-5HPacD";
```
drop these two lines from dtsi.

SvenRoederer replied:
Thanks, missed that during copying ...

rogerpueyo commented:
Hi @SvenRoederer,

I currently have the device installed on a mast which is difficult to reach. I had plans to replace it in the coming weeks to be able to work on it and have it supported with ath79, so it's really good that you've started working on it already. :+1: When I have physical access to it I'll test it and let you know.

rogerpueyo commented on target/linux/ath79/image/mikrotik.mk:
```diff
@@ -11,6 +11,18 @@ define Device/mikrotik_routerboard-493g
+define Device/mikrotik_routerboard-921gs-5hpacd-15s
+  $(Device/mikrotik)
+  SOC := qca9558
+  DEVICE_MODEL := RouterBOARD 921GS-5HPacD-15s
```
On [MikroTik's page](https://mikrotik.com/product/RB921GS-5HPacD-15S), the device is called "mANTBox 15s". We could, maybe, use this as the name, plus the product code "RB921GS-5HPacD-15S" in brackets, i.e., `mANTBox 15s (RB921GS-5HPacD-15S)`. "mANTBox" is easier to read and pronounce, and clearer than 921GS-5HPacD-15s.

Thus, [this one](https://github.com/openwrt/openwrt/pull/3140/commits/d3b63682044d12f6e4cf40d444ef2c20c88bcac6#diff-0b7c4fe8bca5c87d97e92ea6f4757fc9R41) should be changed.

SvenRoederer replied:
Just had a look on some other boards, eems the naming is inconsistent anyway. "Mikrotik hEX" is named "MikroTik RB750Gr3" which is the productID. IMHO making this unique should be put in a separate issue anyway.

f00b4r0 replied:
I think that’s another point for going with combined images. At least per platform. My 2c :)

SvenRoederer replied:
@rogerpueyo  according to "RouterBOARD wAP G-5HacT2HnD (wAP AC)" I changed the name to "RouterBOARD 921GS-5HPacD-15s (mANTBox 15s)"

SvenRoederer replied:
> I think that’s another point for going with combined images. At least per platform. My 2c :)

Probably there will be one day an easy to maintain solution for combined images ...

SvenRoederer commented:
Today I had time to unscrew the chassis, solder the pads and hookup the serial cable.
Sadly the Bootloader seems not to initialize the serial-port, so no output here. But as soon as Linux boots the console gets activated.

When doing the sysupgrade I get following output:
```
root@OpenWrt:/tmp# sysupgrade openwrt-ath79-mikrotik-mikrotik_routerboard-921gs-5hpacd-15s-squashfs-sysupgrade.bin
Cannot save config while running from ramdisk.
Commencing upgrade. Closing all shell sessions.
Watchdog handover: fd=3
- watchdog -
killall: telnetd: no process killed
Sending TERM to remaining processes ... ntpd dnsmasq ubusd urngd logd hostapd netifd odhcpd 
Sending KILL to remaining processes ... 
Performing system upgrade...
Could not open mtd device: firmware
Can't open device for writing!
cat: write error: Broken pipe
[  438.099038] reboot: Restarting system
```

So looks like the script want's to write to the "firmware" mtd-partition which is not defined. I'll rebase from 3f27a6e6402ba52032def578f0944287154a490d to some recent master and retry ...

SvenRoederer commented:
found the missing piece in target/linux/ath79/mikrotik/base-files/lib/upgrade/platform.sh ...

upgrade log:
```
Watchdog handover: fd=3
- watchdog -
killall: telnetd: no process killed
killall: ash: no process killed
Sending TERM to remaining processes ... ntpd dnsmasq ubusd askfirst urngd logd hostapd netifd odhcpd 
Sending KILL to remaining processes ... 
Performing system upgrade...
Unlocking kernel ...
Erasing kernel ...
Writing data to block 0 at offset 0x0
Writing data to block 1 at offset 0x20000
Writing data to block 2 at offset 0x40000
Writing data to block 3 at offset 0x60000
Writing data to block 4 at offset 0x80000
Writing data to block 5 at offset 0xa0000
Writing data to block 6 at offset 0xc0000
Writing data to block 7 at offset 0xe0000
Writing data to block 8 at offset 0x100000
Writing data to block 9 at offset 0x120000
Writing data to block 10 at offset 0x140000
Writing data to block 11 at offset 0x160000
Writing data to block 12 at offset 0x180000
Writing data to block 13 at offset 0x1a0000
Writing data to block 14 at offset 0x1c0000
Writing data to block 15 at offset 0x1e0000
[ 1090.037496] ubi0: attaching mtd7
[ 1090.826558] ubi0: scanning is finished
[ 1090.830664] ubi0 error: ubi_read_volume_table: the layout volume was not found
[ 1090.838258] ubi0 error: ubi_attach_mtd_dev: failed to attach mtd7, error -22
ubiattach: error!: cannot attach mtd7
           error 22 (Invalid argument)
ubiformat: mtd7 (nand), size 130023424 bytes (124.0 MiB), 992 eraseblocks of 131072 bytes (128.0 KiB), min. I/O size 2048 bytes
libscan: scanning eraseblock 991 -- 100 % complete  
ubiformat: 889 eraseblocks are supposedly empty
ubiformat: warning!: 103 of 992 eraseblocks contain non-UBI data
ubiformat: warning!: only 0 of 992 eraseblocks have valid erase counter
ubiformat: erase counter 0 will be used for all eraseblocks
ubiformat: note, arbitrary erase counter value may be specified using -e option
ubiformat: use erase counter 0 for all eraseblocks
ubiformat: format[ 1100.318766] ubi0: attaching mtd7e  
ting eraseblock 991 -- 100 % complete  
[ 1101.746543] ubi0: scanning is finished
[ 1101.766648] ubi0: attached mtd7 (name "ubi", size 124 MiB)
[ 1101.772262] ubi0: PEB size: 131072 bytes (128 KiB), LEB size: 129024 bytes
[ 1101.779249] ubi0: min./max. I/O unit sizes: 2048/2048, sub-page size 512
[ 1101.786050] ubi0: VID header offset: 512 (aligned 512), data offset: 2048
[ 1101.792939] ubi0: good PEBs: 992, bad PEBs: 0, corrupted PEBs: 0
[ 1101.799038] ubi0: user volume: 0, internal volumes: 1, max. volumes count: 128
[ 1101.806370] ubi0: max/mean erase counter: 0/0, WL threshold: 4096, image sequence number: 582584177
[ 1101.815547] ubi0: available PEBs: 968, total reserved PEBs: 24, PEBs reserved for bad PEB handling: 20
[ 1101.825016] ubi0: background thread "ubi_bgt0d" started, PID 1511
UBI device number 0, total 992 LEBs (127991808 bytes, 122.0 MiB), available 968 LEBs (124895232 bytes, 119.1 MiB), LEB size 129024 bytes (126.0 KiB)
Volume ID 0, size 16 LEBs (2064384 bytes, 1.9 MiB), LEB size 129024 bytes (126.0 KiB), dynamic, name "none", alignment 1
Volume ID 1, size 16 LEBs (2064384 bytes, 1.9 MiB), LEB size 129024 bytes (126.0 KiB), dynamic, name "rootfs", alignment 1
Set volume size to 120766464
Volume ID 2, size 936 LEBs (120766464 bytes, 115.1 MiB), LEB size 129024 bytes (126.0 KiB), dynamic, name "rootfs_data", alignment 1
sysupgrade successful
umount: can't unmount /dev: Resource busy
umount: can't unmount /tmp: Resource [ 1103.537199] reboot: Restarting system
```

SvenRoederer commented:
@adrianschmutzler @rogerpueyo changed it to a real PR, ready to merge.

Tested the whole cycle: vendor --> initrd-image --> sysupgrade --> sysupgrade --> vendor

f00b4r0 commented:
@SvenRoederer Can you post full dmesg please? Thanks

f00b4r0 commented on target/linux/ath79/dts/qca9558_mikrotik_routerboard-92x.dtsi:
```diff
@@ -0,0 +1,141 @@
+			};
+
+			bios {
+				read-only;
```
For the sake of consistency I would eventually add `size <0x1000>` here.

adrianschmutzler replied:
Since this is only copied here, I'd do that separately.

SvenRoederer replied:
@f00b4r0 as said in the PR-message / commit-message the dtsi file is make by copying from the RB922.dts file without any change. If this parameter is missing a separate PR is more natural.

f00b4r0 replied:
The point is that not using the `size` keyword was OK-ish in a DTS file dedicated to a specific device where the size of the flash was known to "work" with this parameter missing as all partitions were essentially concatenated. Still, it really should be there and if this partition scheme is used (now via this DTSI) in a device where the flash chip is larger, things will break.

Please do fix this here.

@adrianschmutzler that's the reason (moving from DTS to DTSI) why I'm specifically requesting this be added now, I think this is acceptable.

adrianschmutzler replied:
@f00b4r0 I will not change this a part of this patch, but if you just provide a separate patch I will throw it on top (via GitHub, patchwork, or directly to my mail address)

f00b4r0 replied:
@adrianschmutzler i can't do that while at work, but eventually I would only submit a one-liner exactly as suggested in my first comment. Please do feel free to add it in your tree, I'm not claiming any credit on such a trivial patch.

The current situation is in violation of the DTS specification for routerbootpart (as defined in the driver header). Thanks!

f00b4r0 replied:
Rephrasing: not in violation, rather it opens a door that shouldn't be opened in a DTSI, where this definition can result in an incorrectly sized bios partition. While this is not lethal and mostly cosmetic, I'd rather avoid having this copy-pasted by other contributors as-is. HTH

adrianschmutzler replied:
@f00b4r0 Maybe just paste a commit title/message here, and I will take care of the content.

f00b4r0 replied:
ath79: routerboard-92x: ensure explicit bios size in DTSI

The missing "size" property was acceptable in the context of a single DTS as the underlying device is known to have a 64KB flash, and thus the bios partition fit exactly between the preceding and following ones.

However as this block has moved in a DTSI, for the sake of clarity and explicitness the size argument is added to ensure that if the flash happens to be larger than expected, the bios partition remains properly sized.

adrianschmutzler replied:
Merged in 6e7f40dff3aa3fdbc012d0d9f79a3decdbe3735f.

adrianschmutzler commented on target/linux/ath79/image/mikrotik.mk:
```diff
@@ -11,6 +11,18 @@ define Device/mikrotik_routerboard-493g
+	nand-utils
+  SUPPORTED_DEVICES += rb-921gs-5hpacd-r2
+endef
+TARGET_DEVICES += mikrotik_routerboard-921gs-5hpacd-15s
```
I've just pushed an update to introduce a mikrotik_nand definition. So, this just be reduced to:

```
define Device/mikrotik_routerboard-921gs-5hpacd-15s
  $(Device/mikrotik_nand)
  SOC := qca9558
  DEVICE_MODEL := RouterBOARD 921GS-5HPacD-15s (mANTBox 15s)
  DEVICE_PACKAGES += kmod-ath10k-ct ath10k-firmware-qca988x-ct
  SUPPORTED_DEVICES += rb-921gs-5hpacd-r2
endef
TARGET_DEVICES += mikrotik_routerboard-921gs-5hpacd-15s
```

SvenRoederer replied:
PR has been updated.

adrianschmutzler commented:
Content seems okay from just looking at it.

But I really wonder whether we shouldn't just call the bloody thing mantbox 15s (keeping DTSI name as you did it):

	compatible = "mikrotik,mantbox-15s", "qca,qca9558";
	model = "MikroTik mANTBox 15s";

SvenRoederer commented:
I agree that the name is not really handy. But checking the tree, that's the form used for all mikrotik-boards. To keep consistency we should go as it is.

~~~
$ grep -rn "compatible = \"mikrotik," target/linux/
ath79/dts/ar7161_mikrotik_routerboard-493g.dts:7:       compatible = "mikrotik,routerboard-493g", "qca,ar7161";
ath79/dts/qca9556_mikrotik_routerboard-wap-g-5hact2hnd.dts:10:  compatible = "mikrotik,routerboard-wap-g-5hact2hnd", "qca,qca9556";
ath79/dts/qca9558_mikrotik_routerboard-921gs-5hpacd-15s.dts:7:  compatible = "mikrotik,routerboard-921gs-5hpacd-15s", "qca,qca9558";
ath79/dts/qca9558_mikrotik_routerboard-922uags-5hpacd.dts:7:    compatible = "mikrotik,routerboard-922uags-5hpacd", "qca,qca9558";
ramips/dts/mt7621_mikrotik_routerboard-750gr3.dts:10:   compatible = "mikrotik,routerboard-750gr3", "mediatek,mt7621-soc";
ramips/dts/mt7621_mikrotik_routerboard-m11g.dts:9:      compatible = "mikrotik,routerboard-m11g", "mediatek,mt7621-soc";
ramips/dts/mt7621_mikrotik_routerboard-m33g.dts:9:      compatible = "mikrotik,routerboard-m33g", "mediatek,mt7621-soc";
~~~

Based on this list and the comments here regarding naming I can draft a new PR for changing the displayed name and probably the compatible-string.

SvenRoederer commented:
> @SvenRoederer Can you post full dmesg please? Thanks

@f00b4r0 a initrd-version will be enough?

f00b4r0 commented:
> > Can you post full dmesg please? Thanks
> 
> @f00b4r0 a initrd-version will be enough?

It'll do, thanks.

SvenRoederer commented:
@adrianschmutzler @rogerpueyo  please look at PR #3270 regarding the naming.

adrianschmutzler commented:
In order to get the ath79/mikrotik subtarget forward, I've pulled this into my staging tree here: https://git.openwrt.org/?p=openwrt/staging/adrian.git;a=shortlog;h=refs/heads/mikrotik1

I will built-test and probably merge this in an hour or so.

SvenRoederer commented:
@f00b4r0 here the bootlog you requested

```
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    0.000000] Linux version 4.19.123 (sven@strike) (gcc version 8.4.0 (OpenWrt GCC 8.4.0 r13611+1-3f27a6e640)) #0 Mon Jun 22 21:05:51 2020
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] bootconsole [early0] enabled
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] CPU0 revision is: 00019750 (MIPS 74Kc)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] MIPS: machine is MikroTik RouterBOARD 921GS-5HPacD-15s
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] SoC: Qualcomm Atheros QCA9558 ver 1 rev 0
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] Determined physical RAM map:
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000]  memory: 08000000 @ 00000000 (usable)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] Initrd not found or empty - disabling initrd
Tue Jun 23 22:31:24 2020 kern.warn kernel: [    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
Tue Jun 23 22:31:24 2020 kern.warn kernel: [    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] Zone ranges:
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] Movable zone start for each node
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] Early memory node ranges
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
Tue Jun 23 22:31:24 2020 kern.debug kernel: [    0.000000] On node 0 totalpages: 32768
Tue Jun 23 22:31:24 2020 kern.debug kernel: [    0.000000]   Normal zone: 256 pages used for memmap
Tue Jun 23 22:31:24 2020 kern.debug kernel: [    0.000000]   Normal zone: 0 pages reserved
Tue Jun 23 22:31:24 2020 kern.debug kernel: [    0.000000]   Normal zone: 32768 pages, LIFO batch:7
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    0.000000] random: get_random_bytes called from start_kernel+0x90/0x478 with crng_init=0
Tue Jun 23 22:31:24 2020 kern.debug kernel: [    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
Tue Jun 23 22:31:24 2020 kern.debug kernel: [    0.000000] pcpu-alloc: [0] 0
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32512
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    0.000000] Kernel command line: console=ttyS0,115200n8 rootfstype=squashfs,jffs2
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] Writing ErrCtl register=00000000
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] Readback ErrCtl register=00000000
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] Memory: 120568K/131072K available (4416K kernel code, 148K rwdata, 1056K rodata, 3076K init, 204K bss, 10504K reserved, 0K cma-reserved)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] NR_IRQS: 51
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] CPU clock: 720.000 MHz
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 5309056796 ns
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.000007] sched_clock: 32 bits at 360MHz, resolution 2ns, wraps every 5965232126ns
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.008688] Calibrating delay loop... 359.42 BogoMIPS (lpj=718848)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.043590] pid_max: default: 32768 minimum: 301
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.048910] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.056293] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.067774] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.078670] futex hash table entries: 256 (order: -1, 3072 bytes)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.085546] pinctrl core: initialized pinctrl subsystem
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.092151] NET: Registered protocol family 16
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.120992] clocksource: Switched to clocksource MIPS
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.127621] NET: Registered protocol family 2
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.133196] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 4096 bytes)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.141785] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.149569] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.156665] TCP: Hash tables configured (established 1024 bind 1024)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.163879] UDP hash table entries: 256 (order: 0, 4096 bytes)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.170436] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    0.177684] NET: Registered protocol family 1
Tue Jun 23 22:31:24 2020 kern.debug kernel: [    0.182600] PCI: CLS 0 bytes, default 32
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    1.076999] random: fast init done
Tue Jun 23 22:31:24 2020 kern.warn kernel: [    2.624159] Crashlog allocated RAM at address 0x3f00000
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.631138] workingset: timestamp_bits=30 max_order=15 bucket_order=0
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.642943] squashfs: version 4.0 (2009/01/31) Phillip Lougher
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.649480] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.753183] io scheduler noop registered
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.757575] io scheduler deadline registered (default)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.765031] pinctrl-single 1804002c.pinmux: 544 pins, size 68
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.771881] gpio-export gpio-export: 1 gpio(s) exported
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.778495] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.786226] console [ttyS0] disabled
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.790276] 18020000.uart: ttyS0 at MMIO 0x18020000 (irq = 9, base_baud = 2500000) is a 16550A
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.799900] console [ttyS0] enabled
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.807321] bootconsole [early0] disabled
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.823382] m25p80 spi0.0: gd25d05 (64 Kbytes)
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    2.828238] 1 routerbootpart partitions found on MTD device spi0.0
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    2.834528] Creating 1 MTD partitions on "spi0.0":
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    2.839399] 0x000000000000-0x000000010000 : "partitions"
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    2.846052] 4 routerbootpart partitions found on MTD device partitions
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    2.852721] Creating 4 MTD partitions on "partitions":
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    2.857960] 0x000000000000-0x00000000c000 : "routerboot"
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    2.864035] 0x00000000c000-0x00000000d000 : "hard_config"
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    2.870215] 0x00000000d000-0x00000000e000 : "bios"
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    2.875771] 0x00000000e000-0x00000000f000 : "soft_config"
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.885355] nand: device found, Manufacturer ID: 0x9b, Chip ID: 0xf1
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.891844] nand: ATO NAND 128MiB 3,3V 8-bit
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.896190] nand: 128 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.903894] Scanning device for bad blocks
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    2.952181] 3 fixed-partitions partitions found on MTD device ar934x-nand
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    2.959087] Creating 3 MTD partitions on "ar934x-nand":
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    2.964407] 0x000000000000-0x000000040000 : "booter"
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    2.970126] 0x000000040000-0x000000400000 : "kernel"
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    2.975869] 0x000000400000-0x000008000000 : "ubi"
Tue Jun 23 22:31:24 2020 kern.info kernel: [    2.982628] libphy: Fixed MDIO Bus: probed
Tue Jun 23 22:31:24 2020 kern.err kernel: [    2.987359] ag71xx 19000000.eth: invalid MAC address, using random address
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.310775] libphy: ag71xx_mdio: probed
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.330424] ag71xx 19000000.eth: connected to PHY at mdio.0:04 [uid=004dd074, driver=Atheros 8031 ethernet]
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.340900] eth0: Atheros AG71xx at 0xb9000000, irq 4, mode: rgmii
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.351538] MikroTik RouterBOARD hardware configuration sysfs driver v0.04
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.360284] MikroTik RouterBOARD software configuration sysfs driver v0.02
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.368202] NET: Registered protocol family 10
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.374681] Segment Routing with IPv6
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.378546] NET: Registered protocol family 17
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.383120] 8021q: 802.1Q VLAN Support v1.8
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.389817] PCI host bridge /ahb/pcie-controller@180c0000 ranges:
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.396067]  MEM 0x0000000010000000..0x0000000011ffffff
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.401382]   IO 0x0000000000000000..0x0000000000000000
Tue Jun 23 22:31:24 2020 kern.warn kernel: [    3.406698] ar724x-pci 180c0000.pcie-controller: PCIe link is down
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.413140] PCI host bridge to bus 0000:00
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.417334] pci_bus 0000:00: root bus resource [mem 0x10000000-0x11ffffff]
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.424312] pci_bus 0000:00: root bus resource [io  0x0000]
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.429971] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.436862] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
Tue Jun 23 22:31:24 2020 kern.debug kernel: [    3.445769] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 00
Tue Jun 23 22:31:24 2020 kern.err kernel: [    3.447656] UBI error: no valid UBI magic found inside mtd7
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.467083] Freeing unused kernel memory: 3076K
Tue Jun 23 22:31:24 2020 kern.warn kernel: [    3.471706] This architecture does not have kernel memory protection.
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.478241] Run /init as init process
Tue Jun 23 22:31:24 2020 user.info kernel: [    3.493854] init: Console is alive
Tue Jun 23 22:31:24 2020 user.info kernel: [    3.497555] init: - watchdog -
Tue Jun 23 22:31:24 2020 user.info kernel: [    3.515071] kmodloader: loading kernel modules from /etc/modules-boot.d/*
Tue Jun 23 22:31:24 2020 user.info kernel: [    3.524789] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
Tue Jun 23 22:31:24 2020 user.info kernel: [    3.542529] init: - preinit -
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    3.810159] random: jshn: uninitialized urandom read (4 bytes read)
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    3.857145] random: jshn: uninitialized urandom read (4 bytes read)
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    3.897771] random: jshn: uninitialized urandom read (4 bytes read)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    3.934606] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Tue Jun 23 22:31:24 2020 kern.info kernel: [    7.014433] eth0: link up (1000Mbps/Full duplex)
Tue Jun 23 22:31:24 2020 kern.info kernel: [    7.019168] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Tue Jun 23 22:31:24 2020 kern.info kernel: [    8.064909] eth0: link down
Tue Jun 23 22:31:24 2020 user.info kernel: [    8.077824] procd: - early -
Tue Jun 23 22:31:24 2020 user.info kernel: [    8.080832] procd: - watchdog -
Tue Jun 23 22:31:24 2020 user.info kernel: [    8.635453] procd: - watchdog -
Tue Jun 23 22:31:24 2020 user.info kernel: [    8.638938] procd: - ubus -
Tue Jun 23 22:31:24 2020 kern.warn kernel: [    8.647769] urandom_read: 1 callbacks suppressed
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    8.647777] random: ubusd: uninitialized urandom read (4 bytes read)
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    8.692156] random: ubusd: uninitialized urandom read (4 bytes read)
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    8.699035] random: ubusd: uninitialized urandom read (4 bytes read)
Tue Jun 23 22:31:24 2020 user.info kernel: [    8.706435] procd: - init -
Tue Jun 23 22:31:24 2020 user.info kernel: [    9.136757] kmodloader: loading kernel modules from /etc/modules.d/*
Tue Jun 23 22:31:24 2020 kern.info kernel: [    9.181626] Loading modules backported from Linux version v5.7-rc3-0-g6a8b55ed4056
Tue Jun 23 22:31:24 2020 kern.info kernel: [    9.189347] Backport generated by backports.git v5.7-rc3-1-0-gc0c7d2bb
Tue Jun 23 22:31:24 2020 kern.info kernel: [    9.248162] xt_time: kernel timezone is -0000
Tue Jun 23 22:31:24 2020 kern.info kernel: [    9.367491] PPP generic driver version 2.4.2
Tue Jun 23 22:31:24 2020 kern.info kernel: [    9.381899] NET: Registered protocol family 24
Tue Jun 23 22:31:24 2020 user.info kernel: [    9.457342] kmodloader: done loading kernel modules from /etc/modules.d/*
Tue Jun 23 22:31:24 2020 user.info kernel: [    9.465457] urngd: v1.0.2 started.
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    9.594255] random: crng init done
Tue Jun 23 22:31:24 2020 kern.notice kernel: [    9.597729] random: 3 urandom warning(s) missed due to ratelimiting
Tue Jun 23 22:31:24 2020 user.notice dnsmasq: DNS rebinding protection is active, will discard upstream RFC1918 responses!
Tue Jun 23 22:31:24 2020 user.notice dnsmasq: Allowing 127.0.0.0/8 responses
Tue Jun 23 22:31:24 2020 daemon.info dnsmasq[788]: Connected to system UBus
Tue Jun 23 22:31:24 2020 daemon.info dnsmasq[788]: started, version 2.81 cachesize 150
Tue Jun 23 22:31:24 2020 daemon.info dnsmasq[788]: DNS service limited to local subnets
Tue Jun 23 22:31:24 2020 daemon.info dnsmasq[788]: compile time options: IPv6 GNU-getopt no-DBus UBus no-i18n no-IDN DHCP no-DHCPv6 no-Lua TFTP no-conntrack no-ipset no-auth no-DNSSEC no-ID loop-detect inotify dumpfile
Tue Jun 23 22:31:24 2020 daemon.info dnsmasq[788]: UBus support enabled: connected to system bus
Tue Jun 23 22:31:24 2020 daemon.info dnsmasq[788]: using only locally-known addresses for domain test
Tue Jun 23 22:31:24 2020 daemon.info dnsmasq[788]: using only locally-known addresses for domain onion
Tue Jun 23 22:31:24 2020 daemon.info dnsmasq[788]: using only locally-known addresses for domain localhost
Tue Jun 23 22:31:24 2020 daemon.info dnsmasq[788]: using only locally-known addresses for domain local
Tue Jun 23 22:31:24 2020 daemon.info dnsmasq[788]: using only locally-known addresses for domain invalid
Tue Jun 23 22:31:24 2020 daemon.info dnsmasq[788]: using only locally-known addresses for domain bind
Tue Jun 23 22:31:24 2020 daemon.info dnsmasq[788]: using only locally-known addresses for domain lan
Tue Jun 23 22:31:24 2020 daemon.warn dnsmasq[788]: no servers found in /tmp/resolv.conf.d/resolv.conf.auto, will retry
Tue Jun 23 22:31:24 2020 daemon.info dnsmasq[788]: read /etc/hosts - 4 addresses
Tue Jun 23 22:31:24 2020 daemon.info dnsmasq[788]: read /tmp/hosts/dhcp.cfg01411c - 0 addresses
Tue Jun 23 22:31:46 2020 authpriv.info dropbear[834]: Not backgrounding
Tue Jun 23 22:31:47 2020 user.notice : Added device handler type: 8021ad
Tue Jun 23 22:31:47 2020 user.notice : Added device handler type: 8021q
Tue Jun 23 22:31:47 2020 user.notice : Added device handler type: macvlan
Tue Jun 23 22:31:47 2020 user.notice : Added device handler type: veth
Tue Jun 23 22:31:47 2020 user.notice : Added device handler type: bridge
Tue Jun 23 22:31:47 2020 user.notice : Added device handler type: Network device
Tue Jun 23 22:31:47 2020 user.notice : Added device handler type: tunnel
Tue Jun 23 22:31:49 2020 kern.info kernel: [   37.599536] br-lan: port 1(eth0) entered blocking state
Tue Jun 23 22:31:49 2020 kern.info kernel: [   37.604892] br-lan: port 1(eth0) entered disabled state
Tue Jun 23 22:31:49 2020 kern.info kernel: [   37.610462] device eth0 entered promiscuous mode
Tue Jun 23 22:31:49 2020 kern.info kernel: [   37.630390] IPv6: ADDRCONF(NETDEV_UP): br-lan: link is not ready
Tue Jun 23 22:31:49 2020 daemon.notice netifd: Interface 'lan' is enabled
Tue Jun 23 22:31:49 2020 daemon.notice netifd: Interface 'lan' is setting up now
Tue Jun 23 22:31:49 2020 daemon.notice netifd: Interface 'lan' is now up
Tue Jun 23 22:31:49 2020 daemon.notice netifd: Interface 'loopback' is enabled
Tue Jun 23 22:31:49 2020 daemon.notice netifd: Interface 'loopback' is setting up now
Tue Jun 23 22:31:49 2020 daemon.notice netifd: Interface 'loopback' is now up
Tue Jun 23 22:31:49 2020 daemon.notice netifd: Network device 'lo' link is up
Tue Jun 23 22:31:49 2020 daemon.notice netifd: Interface 'loopback' has link connectivity
Tue Jun 23 22:31:50 2020 daemon.info procd: - init complete -
Tue Jun 23 22:31:50 2020 daemon.info urandom_seed[1120]: Seed saved (/etc/urandom.seed)
Tue Jun 23 22:31:50 2020 daemon.err odhcpd[943]: Failed to send to ff02::1%lan@br-lan (Address not available)
Tue Jun 23 22:31:51 2020 user.notice dnsmasq: found already running DHCP-server on interface 'br-lan' refusing to start, use 'option force 1' to override
Tue Jun 23 22:31:51 2020 daemon.info dnsmasq[788]: read /etc/hosts - 4 addresses
Tue Jun 23 22:31:51 2020 daemon.info dnsmasq[788]: read /tmp/hosts/dhcp.cfg01411c - 0 addresses
Tue Jun 23 22:31:52 2020 kern.info kernel: [   40.678431] eth0: link up (1000Mbps/Full duplex)
Tue Jun 23 22:31:52 2020 kern.info kernel: [   40.683311] br-lan: port 1(eth0) entered blocking state
Tue Jun 23 22:31:52 2020 kern.info kernel: [   40.688642] br-lan: port 1(eth0) entered forwarding state
Tue Jun 23 22:31:52 2020 daemon.notice netifd: Network device 'eth0' link is up
Tue Jun 23 22:31:52 2020 kern.info kernel: [   40.696040] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
Tue Jun 23 22:31:52 2020 daemon.notice netifd: bridge 'br-lan' link is up
Tue Jun 23 22:31:52 2020 daemon.notice netifd: Interface 'lan' has link connectivity
Tue Jun 23 22:31:53 2020 daemon.err odhcpd[943]: Failed to send to ff02::1%lan@br-lan (Address not available)
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[788]: exiting on receipt of SIGTERM
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: Connected to system UBus
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: started, version 2.81 cachesize 150
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: DNS service limited to local subnets
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: compile time options: IPv6 GNU-getopt no-DBus UBus no-i18n no-IDN DHCP no-DHCPv6 no-Lua TFTP no-conntrack no-ipset no-auth no-DNSSEC no-ID loop-detect inotify dumpfile
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: UBus support enabled: connected to system bus
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq-dhcp[1294]: DHCP, IP range 192.168.1.100 -- 192.168.1.249, lease time 12h
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: using only locally-known addresses for domain test
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: using only locally-known addresses for domain onion
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: using only locally-known addresses for domain localhost
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: using only locally-known addresses for domain local
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: using only locally-known addresses for domain invalid
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: using only locally-known addresses for domain bind
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: using only locally-known addresses for domain lan
Tue Jun 23 22:31:57 2020 daemon.warn dnsmasq[1294]: no servers found in /tmp/resolv.conf.d/resolv.conf.auto, will retry
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: read /etc/hosts - 4 addresses
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: read /tmp/hosts/dhcp.cfg01411c - 2 addresses
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq-dhcp[1294]: read /etc/ethers - 0 addresses
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: read /etc/hosts - 4 addresses
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq[1294]: read /tmp/hosts/dhcp.cfg01411c - 2 addresses
Tue Jun 23 22:31:57 2020 daemon.info dnsmasq-dhcp[1294]: read /etc/ethers - 0 addresses
```

SvenRoederer commented:
@adschm thanks for adding the size-parameter into the dts, as suggested by @f00b4r0
